src/peerd/runtime.rs

```diff
@@ -44,15 +43,126 @@ use crate::bus::{
+    let connection = PeerConnection::connect_brontozaur(local_node, remote_node_addr)?;
+    debug!("Connected to remote peer: {}", remote_node_addr);
+
+    debug!("Splitting connection into receiver and sender parts");
+    let (mut peer_receiver, mut peer_sender) = connection.split();
+
+    // this is hella hacky, but it serves the purpose of keeping peerd's service
+    // id constant across reconnects: <REMOTE_NODE_ID>:<REMOTE_ADDR> for taker,
+    // <REMOTE_NODE_ID>:<LOCAL_ADDR> for maker
+    // TODO: It is privacy/security critical that once the
+    // connection is encrypted, this should be replaced by a proper handshake.
+    peer_sender.send_message(PeerMsg::Identity(local_node.node_id()))?;
+    debug!(
+        "sent message with local node id {} to the maker",
+        local_node.node_id()
+    );
+    let unmarshaller: Unmarshaller<PeerMsg> = PeerMsg::create_unmarshaller();
+    let msg: &PeerMsg = &*peer_receiver.recv_message(&unmarshaller)?;
+    match msg {
+        PeerMsg::Pong(id) => {
+            debug!("Received the following pong from the maker {:?}", id);
+        }
+        _ => {
+            return Err(Error::Peer(presentation::Error::UnknownDataType));
+        }
+    };
+
+    let tx = ZMQ_CONTEXT.socket(zmq::PUSH)?;
+    tx.connect("inproc://bridge")?;
+
+    let (thread_flag_tx, _thread_flag_rx) = std::sync::mpsc::channel();
+
+    debug!("Starting thread listening for messages from the remote peer");
+    let bridge_handler = PeerReceiverRuntime {
+        internal_identity: peerd_service_id,
+        bridge: esb::Controller::with(
+            map! {
+                ServiceBus::Bridge => esb::BusConfig {
+                    carrier: Carrier::Socket(tx),
+                    router: None,
+                    queued: true,
+                    api_type: ZmqSocketType::Rep,
+                    topic: None,
+                }
+            },
+            BridgeHandler,
+        )?,
+        _thread_flag_rx,
+        awaiting_pong: false,
+    };
+    let unmarshaller: Unmarshaller<PeerMsg> = PeerMsg::create_unmarshaller();
+    let peer_receiver_runtime = peer::Listener::<PeerReceiverRuntime, PeerMsg>::with(
+        peer_receiver,
+        bridge_handler,
+        unmarshaller,
+    );
+    // We use the _thread_flag_rx to determine when the thread had terminated
+    spawn(move || {
+        debug!("entering peerd receiver runtime loop");
+        if let Err(err) = peer_receiver_runtime.try_run_loop() {
+            error!(
+                "Error encountered in peer receiver runtime, receiver runtime is stopped: {}",
+                err
+            );
+        }
+    });
+
+    Ok((peer_sender, thread_flag_tx))
+}
+
+pub fn run_from_connect(
+    config: ServiceConfig,
+    remote_node_addr: NodeAddr,
+    local_socket: Option<InetSocketAddr>,
+    local_node: LocalNode,
+    // TODO: make this an enum instead with a descriptive distinction of listening and connecting to a listener
```

Comment:
Is this TODO still applicable?

Reply:
1d1ec6ec54d8ce2dcd74a0f8e909047e10c03eed